lib/Driver/Job.cpp

```diff
@@ -403,6 +407,30 @@ void Job::printSummary(raw_ostream &os) const {
   os << "}";
 }
 
+static void
+writeResponseFile(llvm::raw_ostream &OS, const llvm::opt::ArgStringList &Arguments) {
+  // wrap arguments in quotes to ensure compatibility with Unix and Windows
+  for (const char *arg : Arguments) {
+    OS << '"';
+    for (; *arg != '\0'; arg++) {
+      if (*arg == '\"' || *arg == '\\') {
+        OS << '\\'; // escape quote and backslash characters
```

Comment:
I keep looking around for an existing function that does this. The closest is `escapeAndPrintString`; any chance we can combine these?